### PR TITLE
[Version] v2.4.11

### PR DESCRIPTION
**Pull request by Herklos**

### CODE REVIEW

File: octobot_tentacles_manager/configuration/tentacles_setup_configuration.py [hunk 1]
```diff
@@ -298,14 +301,32 @@ def _refresh_profile_tentacles_specific_config(self, tentacles_specific_config_f
...
+            self.installation_context[constants.TENTACLE_INSTALLATION_CONTEXT_OCTOBOT_VERSION] = \
+                self._get_installation_context_bot_version()
+
+    @staticmethod
+    def _get_installation_context_bot_version() -> str:
+        try:
+            import octobot.constants as octobot_constants
+            return octobot_constants.LONG_VERSION
+        except ImportError:
+            return constants.TENTACLE_INSTALLATION_CONTEXT_OCTOBOT_VERSION_UNKNOWN
```
Comment (GuillaumeDSM) [inline octobot_tentacles_manager/configuration/tentacles_setup_configuration.py:315]: 👍

File: octobot_tentacles_manager/api/loader.py [hunk 2]
```diff
@@ -21,5 +24,14 @@ def load_tentacles(verbose=True) -> bool:
     return managers.TentaclesSetupManager.is_tentacles_arch_valid(verbose=verbose)
 
 
+def are_tentacles_up_to_date(tentacles_setup_config, bot_version):
+    installation_version = tentacles_setup_config.installation_context.get(
+        constants.TENTACLE_INSTALLATION_CONTEXT_OCTOBOT_VERSION,
+        constants.TENTACLE_INSTALLATION_CONTEXT_OCTOBOT_VERSION_UNKNOWN)
+    if installation_version is constants.TENTACLE_INSTALLATION_CONTEXT_OCTOBOT_VERSION_UNKNOWN:
+        return False
+    return packaging_version.parse(bot_version) <= packaging_version.parse(installation_version)
```
Comment (GuillaumeDSM) [inline octobot_tentacles_manager/api/loader.py:33]: 👍

File: tests/configuration/test_tentacle_configuration.py [hunk 3]
```diff
@@ -83,9 +85,30 @@ async def test_factory_reset_config():
...
+        constants.TENTACLE_INSTALLATION_CONTEXT_OCTOBOT_VERSION:
+            constants.TENTACLE_INSTALLATION_CONTEXT_OCTOBOT_VERSION_UNKNOWN
+    }
+    assert not api.are_tentacles_up_to_date(setup_config,
+                                            constants.TENTACLE_INSTALLATION_CONTEXT_OCTOBOT_VERSION_UNKNOWN)
+    assert not api.are_tentacles_up_to_date(setup_config, '1.0.0')
+    setup_config.installation_context[constants.TENTACLE_INSTALLATION_CONTEXT_OCTOBOT_VERSION] = '2.0.0'
+    assert not api.are_tentacles_up_to_date(setup_config, '2.1.0')
+    assert api.are_tentacles_up_to_date(setup_config, '2.0.0')
+    assert api.are_tentacles_up_to_date(setup_config, '2.0.0b1')
```
Comment (GuillaumeDSM) [inline tests/configuration/test_tentacle_configuration.py:105]: 👍

File: tests/configuration/test_tentacle_configuration.py [hunk 4]
```diff
@@ -83,9 +85,30 @@ async def test_factory_reset_config():
     _cleanup()
 
 
+async def test_fill_tentacle_config():
+    async with aiohttp.ClientSession() as session:
+        await api.install_all_tentacles(_tentacles_local_path(), aiohttp_session=session)
+
+    setup_config = configuration.TentaclesSetupConfiguration()
+    available_tentacle = util.load_tentacle_with_metadata(constants.TENTACLES_PATH)
+    await setup_config.fill_tentacle_config(available_tentacle, constants.TENTACLE_CONFIG_FILE_NAME)
```
Comment (GuillaumeDSM) [inline tests/configuration/test_tentacle_configuration.py:94]: should we also test a success case (something else than `constants.TENTACLE_INSTALLATION_CONTEXT_OCTOBOT_VERSION_UNKNOWN`) to ensure it works ?

**Review by GuillaumeDSM — COMMENTED** [review]
👍

File: tests/configuration/test_tentacle_configuration.py (hunk 4 above)
Comment (Herklos) [reply tests/configuration/test_tentacle_configuration.py:94]: By mocking octobot_constants.Version call ?

Comment (GuillaumeDSM) [reply tests/configuration/test_tentacle_configuration.py:94]: U guess we can mock `_get_installation_context_bot_version`

Comment (Herklos) [reply tests/configuration/test_tentacle_configuration.py:94]: That's true :+1:

File: tests/configuration/test_tentacle_configuration.py [hunk 5]
```diff
@@ -83,9 +86,39 @@ async def test_factory_reset_config():
...
+        await api.install_all_tentacles(_tentacles_local_path(), aiohttp_session=session)
+
+    setup_config = configuration.TentaclesSetupConfiguration()
+    available_tentacle = util.load_tentacle_with_metadata(constants.TENTACLES_PATH)
+    with mock.patch.object(setup_config, "_get_installation_context_bot_version", mock.Mock()) as bot_version_mock:
+        bot_version_mock.return_value = "1.0.5"
+        await setup_config.fill_tentacle_config(available_tentacle, constants.TENTACLE_CONFIG_FILE_NAME)
+        assert setup_config.installation_context == {
+            constants.TENTACLE_INSTALLATION_CONTEXT_OCTOBOT_VERSION: "1.0.5"
+        }
```
Comment (GuillaumeDSM) [inline tests/configuration/test_tentacle_configuration.py:100]: 👍

**Review by GuillaumeDSM — APPROVED** [review]

**Pull request merged**